File: examples/weather/src/main/java/com/example/weather/spi/WeatherProvider.java
/*
 * Copyright (c) 2026 Konstantin Pavlov and contributors.
 */

package com.example.weather.spi;

import java.util.concurrent.CompletableFuture;

public interface WeatherProvider {

    WeatherObservation currentWeather(String city) throws Exception;

    CompletableFuture<WeatherObservation> currentWeatherAsync(String city);

}

File: examples/weather-mcp-kotlin/src/main/kotlin/com/example/weather/spi/WeatherProvider.kt
/*
 * Copyright (c) 2026 Konstantin Pavlov/IT Staff and contributors.
 */

package com.example.weather.spi

import com.example.weather.model.TemperatureUnit

fun interface WeatherProvider {
    fun currentWeather(
        city: String,
        temperatureUnit: TemperatureUnit,
    ): WeatherObservation
}
